sway/commands/exec_always.c

```diff
@@ -90,8 +108,13 @@ struct cmd_results *cmd_exec_process(int argc, char **argv) {
 	waitpid(pid, NULL, 0);
 	if (child > 0) {
 		sway_log(SWAY_DEBUG, "Child process created with pid %d", child);
-		root_record_workspace_pid(child);
+		if (ctx != NULL) {
+			sway_log(SWAY_DEBUG, "Recording workspace for process %d", child);
+			ctx->pid = child;
+			wl_list_insert(&server.launcher_ctxs, &ctx->link);
```

Comment:
Can we move this to `launcher_ctx_create()`? It doesn't seem like there's a case where `launcher_ctx_create()` is called without inserting in the list?

Reply:
Yep, seems good.